### PR TITLE
docs: add recipe for Read Contracts

### LINKED ISSUE #22: Add recipe for Read Contracts
## Issue Overview

Add a recipe tutorial on reading a contract from the network.

## Proposed Solutions or Ideas

- Your documentation should be clear and concise.
- Use images where appropriate
- Use proper styling and hierarchy for text
- Use proper styling for code and variable names.
- Use proper callouts when required

Refer to docusaurus for more info: https://docusaurus.io/

### PR DESCRIPTION
**Pull request by od-hunter**
# Add recipe for Read Contract

Fixes #22 

## Types of change

- [ ] Bug
- [x] Enhancement
- [x] Documentation

### CODE REVIEW

**Comment by od-hunter** [comment]
@metalboyrick  please review

**Comment by Nadai2010** [comment]
Hi @od-hunter, I really liked your documentation, we look forward to the changes and final review. Thanks

File: docs/recipes/ReadContractDataFromNetwork.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+---
+sidebar_position: 6
+title: Read a uint from a contract
```
Comment (metalboyrick) [inline docs/recipes/ReadContractDataFromNetwork.md:3]: Wrong title

File: docs/recipes/ReadContractDataFromNetwork.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        <div className="text-sm font-semibold">
+          Data: <span>{contractData ? contractData.toString() : "No data available"}</span>
+        </div>
+      )}
+    </div>
+  );
+};
+```
+
+By following this guide, you have successfully implemented a contract reader in your dApp that fetches and displays data from a deployed contract using the [`useScaffoldReactContract`](https://github.com/Scaffold-Stark/scaffold-stark-2/blob/main/packages/nextjs/hooks/scaffold-stark/useScaffoldReadContract.ts) hook.
```
Comment (metalboyrick) [inline docs/recipes/ReadContractDataFromNetwork.md:144]: Typo (useScaffoldReactContract should be useScaffoldReadContract)

**Review by metalboyrick — CHANGES_REQUESTED** [review]
Minor changes and you're good!

File: docs/recipes/ReadContractDataFromNetwork.md
Comment (metalboyrick) [inline docs/recipes/ReadContractDataFromNetwork.md:1]: Please keep the convention consistent with the other files!

**Review by metalboyrick — CHANGES_REQUESTED** [review]

**Comment by od-hunter** [comment]
@metalboyrick @Nadai2010, thank you both for your review.

@metalboyrick, I wasn’t entirely clear on the 'convention consistency' feedback, but I’ve made some recent changes that I hope align with your expectations. If not, I would appreciate further clarification to ensure I implement the conventions consistently.

File: docs/recipes/ReadContractDataFromNetwork.md
Comment (metalboyrick) [reply docs/recipes/ReadContractDataFromNetwork.md:1]: @od-hunter I meant the file name. My bad for the unclear message.

`ReadContractDataFromNetwork.md` --> `read-contract-data.md`

(would prefer this shorter name too)

**Review by Nadai2010 — APPROVED** [review]

**Pull request merged**